### ISSUE TITLE
Compatibility issue with awkward 2.0 and numba

### ISSUE THREAD

**Issue by jpata** [scikit-hep/vector#303]
### Vector Version

0.10.0

### Python Version

3.8.10

### OS / Environment

Ubuntu 20.04

### Describe the bug

The following no longer works under

vector 0.10.0
awkward 2.0.2
numba 0.56.4

while it did use to work under awkward 1.10 (see logs below).

```python
import vector
import awkward as ak
import numba
print("vector", vector.__version__)
print("awkward", ak.__version__)
print("numba", numba.__version__)

vec1 = vector.arr(ak.zip({"pt": [[1.0]], "eta": [[1.0]], "phi": [[1.0]], "energy": [[1.0]]}))
vec2 = vector.arr(ak.zip({"pt": [[1.0, 2.0]], "eta": [[1.0, 2.0]], "phi": [[1.0, 2.0]], "energy": [[1.0, 2.0]]}))

@numba.njit
def match_jets(jets1, jets2):
    assert(len(jets1)==len(jets2))
    for iev in range(len(jets1)):
        j1 = jets1[iev]
        j2 = jets2[iev]
        for ij1 in range(len(j1)):
            for ij2 in range(len(j2)):
                obj1 = j1[ij1]
                obj2 = j2[ij2]
                dr = obj1.deltaR(obj2)
                print(iev, ij1, ij2, dr)

match_jets(vec1, vec2)
```

### Any additional but relevant log output

Failing log with vector 0.10.0, awkward 2.0.2, numba 0.56.4 and produces an enormous error message (see here for the full message: https://gist.github.com/jpata/9e7e92173001b4e13eebc87cbf25984e)

```
$ singularity exec tf-2.11.0.simg python3 test.py 
vector 0.10.0
awkward 2.0.2
numba 0.56.4
Traceback (most recent call last):
  File "test.py", line 24, in <module>
    match_jets(vec1, vec2, 0.3)
  File "/usr/local/lib/python3.8/dist-packages/numba/core/dispatcher.py", line 468, in _compile_for_args
    error_rewrite(e, 'typing')
  File "/usr/local/lib/python3.8/dist-packages/numba/core/dispatcher.py", line 409, in error_rewrite
    raise e.with_traceback(None)
numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
No implementation of function Function(<built-in function getitem>) found for signature:
 
 >>> getitem(ak.ArrayView(ak.RecordArrayType((ak.NumpyArrayType(array(float64, 1d, C), {}), ak.NumpyArrayType(array(float64, 1d, C), {}), ak.NumpyArrayType(array(float64, 1d, C), {}), ak.NumpyArrayType(array(float64, 1d, C), {})), ('rho', 'phi', 'eta', 't'), {"__record__": "Momentum4D"}), {('*', 'Vector2D'): <class 'vector.backends.awkward.VectorArray2D'>, 'Vector2D': <class 'vector.backends.awkward.VectorRecord2D'>, ('*', 'Vector3D'): <class 'vector.backends.awkward.VectorArray3D'>, 'Vector3D': <class 'vector.backends.awkward.VectorRecord3D'>, ('*', 'Vector4D'): <class 'vector.backends.awkward.VectorArray4D'>, 'Vector4D': <class 'vector.backends.awkward.VectorRecord4D'>
 ...
 CLIPPED
...
    Rejected as the implementation raised a specific error:
      AttributeError: 'RecordArrayType' object has no attribute 'recordlookup'
  raised from /usr/local/lib/python3.8/dist-packages/vector/backends/awkward.py:1637

During: typing of intrinsic-call at test.py (19)

File "test.py", line 19:
def match_jets(jets1, jets2, deltaR_cut):
    <source elided>
            for ij2 in range(len(j2)):
                obj1 = j1[ij1]
                ^

```

This did work under under vector 0.10.0, awkward 1.10.1, numba 0.56.3

```
$ singularity exec tf-2.10.0.simg python3 test.py 
vector 0.10.0
awkward 1.10.1
numba 0.56.3
0 0 0 0.0
0 0 1 1.4142135623730951
```

**Comment by Saransh-cpp** [comment]
Thanks for the report, @jpata! This should be fixed by #284, and the fix will be available in vector's next version. The next version might happen after the holidays. If the fix is urgent, you can install the awkward v1+v2 compatible vector using -
```
pip install git+https://github.com/Saransh-cpp/vector@awkward-v2
```

**Comment by jpata** [comment]
Thanks, no rush! I mainly wanted to make sure it's documented (also in case someone ends up here with a google search), and good to hear it has already been fixed in the next release.
